corehq/messaging/smsbackends/infobip/models.py

```diff
@@ -145,5 +157,41 @@ def generate_template_string(cls, template):
 
         template_text = template.get("body", "")
         num_params = template_text.count("{") // 2  # each parameter is bracketed by {{}}
-        parameters = ",".join([f"{{var{i}}}" for i in range(1, num_params + 1)])
+        parameters = ",".join(f"{{var{i}}}" for i in range(1, num_params + 1))
         return f"{WA_TEMPLATE_STRING}:{template['name']}:{template['language']}:{parameters}"
+
+    def get_message(self, backend_message_id):
+        try:
+            config = self.config
+            api_channel = '/sms/1'
+            api_suffix = '/reports'
+            if config.scenario_key:
+                api_channel = '/omni/1'
+
+            headers = {
+                'Authorization': f'App {config.auth_token}',
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            }
+            parameters = {
+                'messageId': backend_message_id
+            }
+            messages = self._get_message_details(api_channel, api_suffix, config, headers, parameters)
+            if not messages:
+                api_suffix = '/logs'
+                messages = self._get_message_details(api_channel, api_suffix, config, headers, parameters)
+            return messages[0]
+        except Exception as e:
+            raise RetryBillableTaskException(str(e))
+
+    def _get_message_details(self, api_channel, api_suffix, config, headers, parameters):
+        url = f'https://{config.personalized_subdomain}.{INFOBIP_DOMAIN}{api_channel}{api_suffix}'
+        response = requests.get(url, params=parameters, headers=headers)
+        response_content = json.loads(response.content)
+        messages = response_content['results']
+        return messages
```

Comment:
requests has built-in support for JSON data
```suggestion
        return response.json()['results']
```